service/src/main/kotlin/fi/espoo/evaka/application/PendingDecisionEmailService.kt

```diff
@@ -0,0 +1,208 @@
+        sendPendingDecisionEmail(db, msg.decisionId)
+    }
+
+    fun scheduleSendPendingDecisionsEmails(db: Database.Connection): Int {
+        val jobCount = db.transaction { tx ->
+            tx.createUpdate("DELETE FROM async_job WHERE type = 'SEND_PENDING_DECISION_EMAIL' AND (claimed_by IS NULL OR completed_at IS NOT NULL)")
+                .execute()
+
+            val pendingDecisions = tx.createQuery(
+                """
+SELECT DISTINCT(d.id)
+FROM decision d
+WHERE d.status = 'PENDING'
+AND d.resolved IS NULL
+AND d.sent_date < current_date - INTERVAL '1 week'
+AND d.pending_decision_emails_sent_count < 2
+AND d.pending_decision_email_sent IS NULL or d.pending_decision_email_sent < current_date - INTERVAL '1 week'
+"""
+            )
+                .mapTo<UUID>()
+                .list()
+
+            logger.info("PendingDecisionEmailService: Scheduling sending ${pendingDecisions.size} pending decision emails")
+
+            pendingDecisions.forEach { decisionId ->
+                asyncJobRunner.plan(
+                    tx,
+                    payloads = listOf(
+                        SendPendingDecisionEmail(
+                            decisionId = decisionId
+                        )
+                    ),
+                    runAt = Instant.now(),
+                    retryCount = 10
+                )
+            }
+
+            pendingDecisions.size
+        }
+
+        asyncJobRunner.scheduleImmediateRun()
+
+        return jobCount
+    }
+
+    fun sendPendingDecisionEmail(db: Database, decisionId: UUID) {
+        try {
+            db.read { tx ->
+                val decision = getDecision(tx.handle, decisionId) ?: run {
+                    throw Exception("Could not send pending decision email for decision $decisionId: decision missing")
+                }
+
+                val application = fetchApplicationDetails(tx.handle, decision.applicationId) ?: run {
+                    throw Exception("Could not send pending decision email for decision ${decision.id}: application could not be found")
+                }
+
+                val guardian = tx.handle.getPersonById(application.guardianId) ?: run {
+                    throw Exception("Could not send pending decision email for application ${application.id}: guardian cannot be found")
+                }
+
+                if (!guardian.email.isNullOrBlank()) {
+                    logger.info("Sending pending decision email to guardian ${guardian.id}")
+
+                    val lang = getLanguage(guardian.language)
+                    emailClient.sendEmail(
+                        decisionId.toString(),
+                        guardian.email,
+                        fromAddress,
+                        getSubject(lang),
+                        getHtml(lang),
+                        getText(lang)
+                    )
+
+                    // Mark as sent
+                    tx.handle.createUpdate(
```

Comment:
You should use `db.transaction` so you can call `createUpdate` on the `Transaction` object directly, using the `handle` to circumvent that is a no-no.

Reply:
Please replace this with `tx.createUpdate`.